src/NamedDims.jl

```diff
@@ -3,13 +3,29 @@ using Base: @propagate_inbounds
 using Base.Broadcast:
     Broadcasted, BroadcastStyle, DefaultArrayStyle, AbstractArrayStyle, Unknown
 using LinearAlgebra
+using Pkg
 using Requires
 using Statistics
 
 export NamedDimsArray, dim, rename
 
 function __init__()
-    @require Tracker = "9f7883ad-71c0-57eb-9f7f-b5c9e6d3789c" include("tracker_compat.jl")
+    @require Tracker = "9f7883ad-71c0-57eb-9f7f-b5c9e6d3789c" include_tracker_compat()
+end
+
+# For Tracker-NamedDims compatibility we require a Tracker version compatible with 0.2.2.
+# Since we use Requires.jl we have no nicer way to set the Tracker compatability.
+function include_tracker_compat()
+    tracker_version = Pkg.installed()["Tracker"]
+    allowed_versions = Pkg.Types.semver_spec("0.2.2")
+    if tracker_version ∈ allowed_versions
+        include("tracker_compat.jl")
+    else
+        @warn string(
+            "Tracker version not compatible with NamedDims. ",
+            "Tracker compatability functionality has been disabled."
+        )
```

Comment:
Why don't you do
```
        @warn string(
            "Tracker version not compatible with NamedDims. ",
            "Tracker compatability functionality has been disabled."
        ) tracker_version allowed_versions
```

Reply:
oops - accidental deletion -- nice catch :)